### PR TITLE
Adding individual block test 

### PR DESCRIPTION
**Pull request by AleksKnezevic**
Added a test to run an individual transformer block from stable diffusion 3.5 medium

### CODE REVIEW

**Comment by kmabeeTT** [comment]
Do we plan to run this in nightly CI at some point (if yes may need to tweak model name to not conflict with existing test)?

**Comment by AleksKnezevic** [comment]
@kmabeeTT, wasn't planning on it, but changed it just in case.

**Review by kmabeeTT — APPROVED** [review]

File: tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+@pytest.mark.parametrize(
+    "mode",
+    ["eval"],
+)
+@pytest.mark.parametrize(
+    "model_info",
+    model_info_list,
+    ids=[model_info[0] for model_info in model_info_list],
+)
+def test_stable_diffusion_transformer(record_property, model_info, mode):
```
Comment (ddilbazTT) [inline tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py:58]: Maybe we should change this name to `test_stable_diffusion_transformer_block`

File: tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# SPDX-FileCopyrightText: (c) 2025 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+import torch
+import pytest
+from tests.utils import ModelTester, skip_full_eval_test
+from diffusers import (
+    StableDiffusion3Pipeline,
```
Comment (ddilbazTT) [inline tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py:8]: nit: we are not using the pipeline or scheduler so we could have removed it for cleanliness

File: tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# SPDX-FileCopyrightText: (c) 2025 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+import torch
+import pytest
+from tests.utils import ModelTester, skip_full_eval_test
```
Comment (ddilbazTT) [inline tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py:6]: nit: we are not using `skip_full_eval_test` I am assuming this runs end-to-end

File: tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            "hidden_states": hidden_states,
+            "encoder_hidden_states": encoder_hidden_states,
+            "temb": temb,
+            "joint_attention_kwargs": joint_attention_kwargs,
+        }
+        return arguments
+
+
+model_info_list = [
+    ("SD3.5-medium-transformer-block", "stabilityai/stable-diffusion-3.5-medium"),
```
Comment (ddilbazTT) [inline tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py:45]: nit: we could add large too

**Review by ddilbazTT — APPROVED** [review]
Test looks good, but why are we adding this? Out of curiosity, since it's really similar to the test we're already running

File: tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py (hunk 4 above)
Comment (AleksKnezevic) [reply tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py:45]: I didn't look up the input shapes for large, I assume they'll be different.

File: tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py (hunk 3 above)
Comment (AleksKnezevic) [reply tests/models/stable_diffusion/test_stable_diffusion_transformer_block.py]: fixed.
